tests/test_export.py

```diff
@@ -222,3 +226,237 @@ def test_expand_env():
+
+
+def test_named_and_nameless():
+    cells = [
+        'a = 1',
+        'b = 2',
+        '# nuclio: end-code my-function',
+        '# nuclio: start-code',
+        'd = 4',
+        'e = 2',
+    ]
+    validate_code(cells[2:], cells[:2], cells)
+
+    validate_code_with_function_name(cells[:2],
+                                     cells[2:],
+                                     cells,
+                                     'my-function')
+
+
+def test_overlap():
+    cells = [
+        '# nuclio: start-code',
+        'a = 1',
+        'b = 2',
+        '# nuclio: end-code my-function',
+        'd = 4',
+        'e = 2',
+    ]
+    validate_code(cells, [], cells)
+
+    validate_code_with_function_name(cells[:3],
+                                     cells[3:],
+                                     cells,
+                                     'my-function')
+
+
+def test_start_code_overlap_named():
+    cells = [
+        'a = 1',
+        '# nuclio: start-code my-function',
+        'b = 2',
+        '# nuclio: start-code another-function',
+        'd = 4',
+        'e = 2',
+        '# nuclio: end-code my-function',
+    ]
+    validate_code_with_function_name(cells[1:],
+                                     [cells[0]],
+                                     cells,
+                                     'my-function')
+    validate_code_with_function_name(cells[3:],
+                                     cells[:3],
+                                     cells,
+                                     'another-function')
+
+
+def test_end_code_overlap_named():
+    cells = [
+        'a = 1',
+        '# nuclio: start-code my-function',
+        'b = 2',
+        '# nuclio: end-code another-function',
+        'd = 4',
+        'e = 2',
+        '# nuclio: end-code my-function',
+    ]
+    validate_code_with_function_name(cells[1:],
+                                     [cells[0]],
+                                     cells,
+                                     'my-function')
+    validate_code_with_function_name(cells[:3],
+                                     cells[3:],
+                                     cells,
+                                     'another-function')
+
+
+def test_overlap_end_code():
+    cells = [
+        'a = 1',
+        '# nuclio: start-code my-function',
+        'b = 2',
+        'd = 4',
+        '# nuclio: end-code',
+        'e = 2',
+        '# nuclio: end-code my-function',
+    ]
+    validate_code_with_function_name(cells[1:],
+                                     [cells[0]],
+                                     cells,
+                                     'my-function')
+
+    validate_code(cells[:4], cells[4:], cells)
+
+
+def validate_code(expected_cells, expected_excluded_cells, cells):
+    nb = gen_nb(cells)
+    code, _ = export_notebook(nb)
+    for cell in expected_cells:
+        cell = export.filter_comments(cell)
+        assert cell in code, '{!r} in code'.format(cell)
```

Comment:
```suggestion
        assert cell in code, '{0!r} not in code'.format(cell)
```
The 0 ordinal shouldn't make any difference here it's just a convention from zebo we use where we do python2 style string formatting.
Also - I'm assuming we need to supposed python2 here, so not suggesting to change to f-strings